core/chain_makers.go

```diff
@@ -108,7 +109,7 @@ func (b *BlockGen) AddTxWithChain(bc *BlockChain, tx *types.Transaction) {
 	b.receipts = append(b.receipts, receipt)
 	if tokenFeeUsed {
 		fee := new(big.Int).SetUint64(gas)
-		if b.header.Number.Cmp(common.TIPTRC21Fee) > 0 {
+		if b.config.IsTIPTRC21Fee(b.header.Number) {
```

Comment:
I doubt that the comparison is incorrect. The hardfork check function use `isForked` internally, which is `s.Cmp(head) <= 0` or head block must be GreaterThanEqual hardfork block. The original statement only have Greater comparison

Reply:
That's correct, I missed the point. It must be checked with `b.header.Number + 1`